lib/utmp.c

```diff
@@ -210,7 +210,7 @@ get_session_host(char **out, pid_t main_pid)
 	ut = get_current_utmp(main_pid);
 
 #if defined(HAVE_STRUCT_UTMPX_UT_HOST)
-	if ((ut != NULL) && (ut->ut_host[0] != '\0')) {
+	if ((ut != NULL) && strncmp(ut->ut_host, "", countof(ut->ut_host)) != 0) {
```

Comment:
I disagree that this is simpler to read.  But also, what's the advantage of
using strncmp here, instead of just strcmp(x, "");

Reply:
If you really want to not see the byte comparisons (... I like them ...) you
could add an is_empty_string() fn?  That would be easier to read.